web_ui/src/pages/user-management/users/actions/edit-organization-user-dialog.component.tsx

```diff
@@ -0,0 +1,230 @@
+import {
+    Button,
+    ButtonGroup,
+    Content,
+    Dialog,
+    Divider,
+    Flex,
+    Form,
+    Heading,
+    TextField,
+    Tooltip,
+    TooltipTrigger,
+} from '@geti/ui';
+import { Email } from '@geti/ui/icons';
+
+import { StatusCell } from '../../../../shared/components/table/status-cell/status-cell.component';
+import { RolePicker } from '../old-project-users/role-picker.component';
+import { OrganizationRoleTooltipContent } from '../organization-role-tooltip/organization-role-tooltip';
+import { LastLoginCell } from '../users-table/last-login-cell.component';
+
+import classes from '../workspace-users/actions/user-summary.module.scss';
+
+interface EditOrganizationUserDialogProps {
+    organizationId: string;
+    user: User;
+    users: User[];
+    activeUser: User;
+    isSaasEnvironment: boolean;
+    closeDialog: () => void;
+}
+
+export const EditOrganizationUserDialog = ({
+    organizationId,
+    user,
+    users,
+    activeUser,
+    isSaasEnvironment,
+    closeDialog,
+}: EditOrganizationUserDialogProps) => {
+    const { FEATURE_FLAG_MANAGE_USERS_ROLES } = useFeatureFlags();
+    const { useUpdateUser, useUpdateUserRoles, useUpdateMemberRole } = useUsers();
+    const updateUser = useUpdateUser();
+    const updateUserRoles = useUpdateUserRoles();
+    const updateMemberRole = useUpdateMemberRole();
+
+    const isActiveOrgAdmin = isOrganizationAdmin(activeUser, organizationId);
+    const isEditingSelf = activeUser.id === user.id;
+
+    const [firstName, setFirstName] = useState(user.firstName);
+    const [lastName, setLastName] = useState(user.lastName);
+
+    const currentOrgRole: USER_ROLE | undefined = useMemo(
+        () =>
+            user.roles.find((r) => r.resourceType === RESOURCE_TYPE.ORGANIZATION && r.resourceId === organizationId)
+                ?.role,
+        [user.roles, organizationId]
+    );
+
+    const [selectedOrgRole, setSelectedOrgRole] = useState<USER_ROLE | undefined>(currentOrgRole);
+
+    const orgAdmins = useMemo(
+        () =>
+            users.filter((u) =>
+                u.roles.some(
+                    (r) =>
+                        r.resourceType === RESOURCE_TYPE.ORGANIZATION &&
+                        r.resourceId === organizationId &&
+                        r.role === USER_ROLE.ORGANIZATION_ADMIN
+                )
+            ),
+        [users, organizationId]
+    );
+    const isLastRemainingOrgAdmin = currentOrgRole === USER_ROLE.ORGANIZATION_ADMIN && orgAdmins.length === 1;
+
+    const canEditNames = isActiveOrgAdmin || isEditingSelf;
+
+    const nameChanged = firstName !== user.firstName || lastName !== user.lastName;
+    const roleChanged = selectedOrgRole !== currentOrgRole;
+
+    const isSaveDisabled = !nameChanged && !roleChanged;
+
+    const handleSubmit = async (e: FormEvent) => {
+        e.preventDefault();
+        const promises: Promise<User | void>[] = [];
+
+        if (nameChanged && !isSaasEnvironment && canEditNames) {
```

Comment:
Move this into a var, then

```suggestion
        if (canUpdateUser) {
```